File: frontend/src/models/stockTransaction.js
/**
 * @typedef {Object} StockTransaction
 * @property {string} id
 * @property {string} productId
 * @property {string} userId
 * @property {'IN' | 'OUT' | 'ADJUSTMENT'} type
 * @property {number} quantity
 * @property {string} reason
 * @property {string} timestamp
 */

export const exampleStockTxn = {
    id: "uuid",
    productId: "uuid",
    userId: "uuid",
    type: "IN",
    quantity: 50,
    reason: "Restock",
    timestamp: "2025-09-23T14:00:00Z"
  };
  

File: backend/src/main/java/com/smartstock/model/StockTransaction.java
package com.smartstock.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "stock_transactions")
public class StockTransaction {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "product_id")
    @NotNull
    private Product product;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @NotNull
    private User user;

    @Enumerated(EnumType.STRING)
    @NotNull
    private StockTransactionType type;

    @Min(1)
    private int quantity;

    private String reason;

    @PastOrPresent
    private LocalDateTime timestamp;
}
